game-ai/src/main/java/org/triplea/ai/flowfield/FlowFieldAiProvider.java

```diff
@@ -0,0 +1,22 @@
+package org.triplea.ai.flowfield;
+
+import games.strategy.engine.framework.startup.ui.PlayerTypes;
+import games.strategy.triplea.ai.AbstractAi;
+import games.strategy.triplea.ai.AiProvider;
+
+public class FlowFieldAiProvider implements AiProvider {
+  @Override
+  public AbstractAi create(final String name, final PlayerTypes.AiType playerType) {
+    return new FlowFieldAi(name, playerType);
+  }
+
+  @Override
+  public String getLabel() {
+    return "FlowField (AI)";
+  }
+
+  @Override
+  public boolean isBetaPlayer() {
```

Comment:
Most AI is not going to be beta, and beta should be a relatively short thing in the long run. After we finish this AI we'll have years and years of no beta players. There is a 'isVisible' field on player type, can we use that to hide beta players?

Reply:
The `isBetaPlayer` is converted into the `isVisible` field after checking the beta features flag.  Renaming it to `isVisible` has the same problems as what you mentioned.  It will be "invisible" for a relatively short time and then years and years of being visible.  So I didn't change the name but I can do that if you want.

Reply:
Hmm. Interesting. I've had conversations with people who believe feature flags are an anti-pattern. This situation reminds me of those conversations.

I do think we should avoid having specific APIs that change when a feature flag is removed. Ideally all that happens is an `if` block is removed and the `else` part of it is also deleted as it's no longer used.

If all player types come from when we instantiate 'injections', we should be able to feature flag filter flow-field at that time, right?

Reply:
Though `ClientSetting` is part of game-core. It really should not be though. Without lifting `ClientSettings` into game-core we might have to do something kinda nasty to get the flagging right and have it live in `game-headed`...

Reply:
`Injections` is instantiated before `ClientSetting` is setup.  So accessing `ClientSetting` during the `Injections` instantiation fails because it isn't setup.

Reply:
AFAIK, client settings can be instantiated first. The init of it does not really depend on anything. Can we swap the order? 

Side-note: I've had thoughts we might want to migrate ClientSettings to be backed by a property file. It could lend to easier code and make the static initialization no longer a concern. It is something to think about how to do that more cleanly (it has scaled pretty well to some extent, though the static access and initialization, particularly in test are painful).

Reply:
I just swapped the order and the game loaded.  I've pushed a change to remove beta ai when `Injections` is created.  This means that you have to quit the game after you change the beta flag to see changes to the list of players.